### PR TITLE
Applications: nrf5340_audio: synchronize L and R with SDC.

### PR DESCRIPTION
**Pull request by henrla**
Use timestamps in SDUs to synchronize the left and right channels. The approach to synchronize the left and right channels are the following. If two channels send their first SDU simultaneously, it is without a timestamp as the controller cannot provide a timestamp value using read_iso_tx_sync until after the first SDU has been sent. The first two SDUs will be given a timestamp by the controller. This timestamp is based on time of arrival. Unfortunate timings can lead to the two timestamps belonging to different CIG reference points. In this situation, read_iso_tx_sync will return different tx timestamps for the two channels. The channel that is behind must use the timestamp of the channel that is in front in time to catch up and synchronize.

### CODE REVIEW

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c [hunk 1]
```diff
@@ -778,7 +779,7 @@ static void stream_sent_cb(struct bt_bap_stream *stream)
 
 	ret = channel_index_get(stream->conn, &channel_index);
 	if (ret) {
-		LOG_ERR("Channel index not found");
+		LOG_WRN("Channel index not found");
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:782]: Why this change?

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c [hunk 2]
```diff
@@ -1306,11 +1308,17 @@ int unicast_client_stop(void)
 	return 0;
 }
 
+
+#define SDU_INTERVAL_US (10000u)
```
Comment (koffes) [inline applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:1312]: I guess this is just for testing as we need to get the actual interval.

File: applications/nrf5340_audio/src/modules/audio_sync_timer.h [hunk 3]
```diff
@@ -17,6 +17,13 @@
  */
 uint32_t audio_sync_timer_capture(void);
 
+/**
+ * @brief Add some documentation :)
+ * 
+ * @retval
+*/
+uint32_t audio_sync_frame_send_timer_capture(void);
```
Comment (koffes) [inline applications/nrf5340_audio/src/modules/audio_sync_timer.h:25]: Never used. Can we omit these two files?

**Review by koffes — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c (hunk 2 above)
Comment (henrla) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:1312]: Correct.

File: applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c (hunk 1 above)
Comment (henrla) [reply applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c:782]: For debugging purposes. Should not be included.

File: applications/nrf5340_audio/src/modules/audio_sync_timer.h (hunk 3 above)
Comment (henrla) [reply applications/nrf5340_audio/src/modules/audio_sync_timer.h:25]: For testing purposes, I didn't have to use it. For the real thing, I think it might have to be used to make the solution more robust, due to how timestamped SDUs works on our controller. I think we should discuss the scenario where I think this is needed elsewhere.

**Comment by koffes** [comment]
Resolved in https://github.com/nrfconnect/sdk-nrf/pull/13589
